core/src/Revolution/modResource.php

```diff
@@ -1537,4 +1539,70 @@ public function clearCache($context = '')
             $this->xpdo->invokeEvent('OnResourceCacheUpdate', ['id' => $this->get('id')]);
         }
     }
+
+    /**
+     * Get icon classes for the resource, based on template, class_key, system settings, etc.
+     *
+     * @return array
+     */
+    public function getIconClasses()
+    {
+        try {
+            $reflectionClass = new ReflectionClass($this->get('class_key'));
+            $classKey = strtolower($reflectionClass->getShortName());
+        } catch (ReflectionException $e) {
+            $classKey = strtolower($this->get('class_key'));
+        }
+
+        if (substr($classKey, 0, 3) === 'mod') {
+            $classKey = substr($classKey, 3);
+        }
+
+        $iconCls = [];
+
+        $contentType = $this->getOne('ContentType');
+        if ($contentType && $contentType->get('icon')) {
+            $iconCls = [$contentType->get('icon')];
```

Comment:
Shouldn't this (and the same logic on line 1572) be `$iconCls[] = $contentType->get('icon')` or is the point that we can only have a single icon between the content type and template?

This does seem to be the same in the original code so happy to let it slide if it's just the way things used to be.

Reply:
Adjusted, a relic from past